java/configuration/src/main/java/sleeper/configuration/properties/SleeperProperty.java

```diff
@@ -19,40 +19,88 @@
 import java.util.function.Predicate;
 
 /**
- * Interface to be implemented by all property enums. Allows SleeperProperties to set generically. Depending on
- * implementation the class may want to provide a means of validating the provided property.
+ * Defines a property used in Sleeper configuration. Used to generate documentation and to validate values and process
+ * changes for a single Sleeper property.
  */
 public interface SleeperProperty {
+
+    /**
+     * Retrieves the name of the property, used to set the value in a properties file.
+     *
+     * @return the property name
+     */
     String getPropertyName();
 
+    /**
+     * Retrieves the default value of the property. May be null if the property has no default value. This is usually
+     * the value the property will have if it is not set, but some properties have other behaviour to compute thier
+     * value, e.g. to retrieve the value from another property.
+     *
+     * @return the default value
+     */
     String getDefaultValue();
 
+    /**
+     * Retrieves a description of the property. Used in documentation and when displaying the value in a human-readable
+     * way.
+     *
+     * @return the description
+     */
     String getDescription();
 
+    /**
+     * Retrieves the property's group. Used to gather related properties together for display.
+     *
+     * @return the group
+     */
     PropertyGroup getPropertyGroup();
 
+    /**
+     * Checks whether the property is set automatically at deploy time by the CDK. The CDK will clear and overwrite all
+     * properties that are set by the CDK, on every deployment.
+     *
+     * @return true if set by the CDK, false if set before CDK deployment
+     */
     default boolean isSetByCdk() {
         return false;
     }
 
+    /**
+     * Checks whether the property is editable. For example, changing the instance ID would mean referring to a
+     * completely different instance of Sleeper, so it doesn't make sense to edit it.
+     *
+     * @return true if the property is editable
+     */
     default boolean isEditable() {
         return !isSetByCdk();
     }
 
+    /**
+     * Checks whether the property may be set by the user. If false, it may be set by the CDK, or by some other
```

Comment:
Definition of false works better in the return section, aligning it with above design (isSetByCDK javadoc)